src/components/signup/SignUpForm.js

```diff
@@ -1,75 +1,127 @@
 import { useState } from "react";
+import axiosInstance from "../../api/axios";
+import { Link } from "react-router-dom";
 import "./SignUpForm.css";
 
 const SignUpForm = () => {
-  const [signUpUser, setSignUpUser] = useState({
+  const [signUpDetails, setSignUpDetails] = useState({
     name: "",
     email: "",
     password1: "",
     password2: "",
-  })
-  const [sucess, setSuccess] = useState(false) // for successful state upon signing up
-  const [error, setError] = useState("") // to show error messages assiciated with signing up
+  });
+  const [sucess, setSuccess] = useState(false); // for successful state upon signing up
+  const [error, setError] = useState(""); // to show error messages assiciated with signing up
+
   const handleSignUpChange = (e) => {
-    setSignUpUser((prev) => ({
-      ...prev, [e.target.value]: e.target.value.trim()
-    }))
-  }
-  
+    setSignUpDetails((prev) => ({
+      ...prev,
+      [e.target.name]: e.target.value,
+    }));
+  };
+
+  const postRequest = async () => {
+    try {
+      await axiosInstance.post("dj-rest-auth/registration/", signUpDetails);
+      setSuccess(true);
+    } catch (e) {
+      setSuccess(false);
+      setError("Registration failed, please try again.");
+      console.log(e);
+    }
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (
+      !signUpDetails.name ||
+      !signUpDetails.email ||
+      !signUpDetails.password1 ||
+      !signUpDetails.password2
+    ) {
+      setError("Please provide details");
+      setSuccess(false);
+    } else if (signUpDetails.password1 !== signUpDetails.password2) {
+      setError("Password does not match");
+      setSuccess(false);
+    } else if (signUpDetails.password1.length <= 7) {
```

Comment:
Line 47's check should come before line 44

Reply:
oh ok. I will adjust this.